### ISSUE TITLE
Expo Linking is using hostUri of null in bare workflow for universal link/deep link usage

### ISSUE THREAD

**Issue by OscarYuen** [expo/expo#8973]
### Summary of Issue (just a few sentences)
Expo Linking package's parse() & makeUri() functions are calling getHostUri() function [https://github.com/expo/expo/blob/bd5675335b969614899d477e50e02443a92b7f29/packages/expo-linking/src/Linking.ts#L24](https://github.com/expo/expo/blob/bd5675335b969614899d477e50e02443a92b7f29/packages/expo-linking/src/Linking.ts#L24)
However, getHostUri() would return the hostUri of Constant manifest which is null in bare worflow if expo-updates is not installed(https://github.com/expo/expo/issues/7721).
The app would crash if parse() & makeUri() are called.

### Environment - output of `expo diagnostics` & the platform(s) you're targeting
Expo CLI 3.21.10 environment info:
    System:
      OS: macOS 10.15.3
      Shell: 5.7.1 - /bin/zsh
    Binaries:
      Node: 12.18.1 - ~/.nvm/versions/node/v12.18.1/bin/node
      Yarn: 1.21.1 - /usr/local/bin/yarn
      npm: 6.14.5 - ~/.nvm/versions/node/v12.18.1/bin/npm
    IDEs:
      Android Studio: 3.5 AI-191.8026.42.35.6010548
      Xcode: 11.5/11E608c - /usr/bin/xcodebuild
    npmPackages:
      expo: ~37.0.3 => 37.0.12 
      react: ~16.9.0 => 16.9.0 
      react-dom: ~16.9.0 => 16.9.0 
      react-native: ~0.61.5 => 0.61.5 
      react-native-web: ~0.11.7 => 0.11.7 
    npmGlobalPackages:
      expo-cli: 3.21.10

### Reproducible Demo
1. Open a new bare workflow template without expo-updates
2. Add the following test code in App.js
[https://snack.expo.io/@oscaryuen/360295](https://snack.expo.io/@oscaryuen/360295)
3. Setup the deep link/universal link in Xcode and Android manifest by iOS/Android official guides
4. Open the deep link/universal link

### Steps to Reproduce
1. The deep link/universal link would make the app open and call Linking.parse to output console log in Expo snack

### Expected Behavior vs Actual Behavior
Expected:
The deep link/universal link can open the app without crashing

Actual Behavior:
The app would crash with the following error message from my Firebase TestLab:
```
Fatal exception
value (address at@1:79217
     FATAL EXCEPTION: mqt_native_modules
Process: co.abcd.abcd, PID: 6517
com.facebook.react.common.JavascriptException: TypeError: Cannot read property 'hostUri' of null, js engine: hermes, stack:
L (address at@1:575743
q (address at@1:576251
Ge (address at@1:1810863
Ae (address at@1:1810833
apply@-1
value (address at@1:85688
apply@-1
value (address at@1:80836
anonymous (address at@1:79259
value (address at@1:80434
value (address at@1:79217
	at com.facebook.react.modules.core.ExceptionsManagerModule.reportException(ExceptionsManagerModule.java:71)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:371)
	at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:150)
	at com.facebook.react.bridge.queue.NativeRunnable.run(Native Method)
	at android.os.Handler.handleCallback(Handler.java:789)
	at android.os.Handler.dispatchMessage(Handler.java:98)
	at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:26)
	at android.os.Looper.loop(Looper.java:251)
	at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:225)
	at java.lang.Thread.run(Thread.java:764)


```

**Comment by sungsong88** [comment]
+1 for this.
It's using `Constants` from 'expo-constants' as `manifest`, and that doesn't have hostUri in bare workflow.

**Comment by esamelson** [comment]
This crash should be fixed by #10195, as well as recent changes in expo-constants 9.3.x which make Constants.manifest defined in the bare workflow.
